openssl/src/cms.rs

```diff
@@ -219,11 +219,61 @@ impl CmsContentInfo {
             Ok(CmsContentInfo::from_ptr(cms))
         }
     }
+
+    /// Verify this CmsContentInfo's signature, given a stack of certificates
+    /// in certs, an X509 store in store. If the signature is detached, the
+    /// data can be passed in data. The data sans signature will be copied
+    /// into output_data if it is present.
+    ///
+    /// OpenSSL documentation at [`CMS_verify`]
+    ///
+    /// [`CMS_verify`]: https://www.openssl.org/docs/manmaster/man3/CMS_verify.html
+    pub fn verify(
+        &mut self,
+        certs: Option<&Stack<X509>>,
+        store: Option<&X509Store>,
+        data: Option<&[u8]>,
+        output_data: Option<&mut Vec<u8>>,
+        flags: CMSOptions,
+    ) -> Result<bool, ErrorStack> {
+        unsafe {
+            let certs = match certs {
+                Some(certs) => certs.as_ptr(),
+                None => ptr::null_mut(),
+            };
+            let store = match store {
+                Some(store) => store.as_ptr(),
+                None => ptr::null_mut(),
+            };
+            let in_data = match data {
+                Some(data) => MemBioSlice::new(data)?.as_ptr(),
+                None => ptr::null_mut(),
+            };
+            let out_bio = MemBio::new()?;
+
+            let is_valid = cvt_n(ffi::CMS_verify(
```

Comment:
The documentation claims that either CMS_verify returns 1 or an error occurs, so it seems like is_valid is not necessary.

Reply:
You are right. There is no difference between validation failure and any other error.